app/src/main/java/com/mr/nemo/dragonfly/ui/component/securitycode/CodeTextField.kt

```diff
@@ -0,0 +1,157 @@
+package com.mr.nemo.dragonfly.ui.component.securitycode
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.aspectRatio
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.OutlinedTextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.drawWithContent
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusRequester
+import androidx.compose.ui.platform.LocalSoftwareKeyboardController
+import androidx.compose.ui.platform.SoftwareKeyboardController
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.core.text.isDigitsOnly
+import com.mr.nemo.dragonfly.ui.component.textfield.BaseTextField
+import com.mr.nemo.dragonfly.ui.theme.DragonFlyTheme
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+fun CodeTextField(
+    digit: String,
+    onValueChanged: (String) -> Unit,
+    currentFocus: FocusRequester,
+    modifier: Modifier = Modifier,
+    size: Dp = 40.dp,
+    previousFocus: FocusRequester? = null,
+    nextFocus: FocusRequester? = null,
+    visualTransformation: VisualTransformation = PasswordVisualTransformation()
+) {
+    val colors = DragonFlyTheme.colors
+    val typography = DragonFlyTheme.typography
+    val keyboardManager = LocalSoftwareKeyboardController.current
+
+    val borderColor by remember {
+        derivedStateOf {
```

Comment:
@bohdanzdaniekin any particular reason to use `derivedStateOf` here?

Reply:
nah 🗿